src/lib/log.c

```diff
@@ -57,12 +66,22 @@ static void log_internal(FILE *out, const char *file, int line, const char *labe
 		if (printed < DEBUG_HEADER_SIZE) {
 			fprintf(out, "%*c", 50 - printed, ' ');
 		}
+
+		/* all debug messages need to finish with a line break, even if the
+		 * user didn't add one to the message, otherwise the output will be
+		 * messed up */
+		if (format[strlen(format) - 1] != '\n') {
+			format_end = "\n";
+		}
 	}
 
 	if (label) {
 		fprintf(out, "%s: ", label);
 	}
 	vfprintf(out, format, args_list);
+	fprintf(out, "%s", format_end);
+
+exit:
 	fflush(out);
```

Comment:
This was added by a previous patch, but adding a fflush() is usually not a good idea because it can slow down swupd when there's too much logs.
What is the problem to remove this?

Reply:
I don't remember why I added it in a previous patch, I remember when it wasn't there the output was not correctly printed. But we could remove it and see how it goes. And add it back if we see is necessary.

Reply:
yes. it would be nice to test that